libraries/AP_GPS/AP_GPS_DroneCAN.cpp

```diff
@@ -381,8 +381,17 @@ void AP_GPS_DroneCAN::handle_fix2_msg(const uavcan_equipment_gnss_Fix2& msg, uin
         loc.lat = msg.latitude_deg_1e8 / 10;
         loc.lng = msg.longitude_deg_1e8 / 10;
         const int32_t alt_amsl_cm = msg.height_msl_mm / 10;
-        interim_state.have_undulation = true;
-        interim_state.undulation = (msg.height_msl_mm - msg.height_ellipsoid_mm) * 0.001;
+        // if ellipsoid height is not supported by the GPS driver (or always
+        // with older releases), AP_Periph reports height_ellipsoid_mm == height_msl_mm .
```

Comment:
Should we stop doing that? Although it looks like there is no way to report as not available in the message.

Reply:
Yes, as mentioned future work is to define a "not available" value and make at least both sides of ArduPilot use it.